src/Storages/ObjectStorage/DataLakes/DeltaLake/TableSnapshot.cpp

```diff
@@ -166,7 +166,7 @@ class TableSnapshot::Iterator final : public DB::IObjectIterator
     {
         if (filter.has_value() && enable_engine_predicate)
         {
-            auto predicate = getEnginePredicate(filter.value(), engine_predicate_exception);
+            auto predicate = getEnginePredicate(filter.value(), engine_predicate_exception, DB::Context::getGlobalContextInstance());
```

Comment:
I see it's too complex to propagate real context here for now, let's not use global context and use `nullptr` in `createInternalCast` in `EnginePredicate.h`